gst/nnstreamer/nnstreamer_plugin_api_filter.h

```diff
@@ -184,41 +252,81 @@ typedef struct _GstTensorFilterFramework
        */
 
-  int (*reloadModel) (const GstTensorFilterProperties * prop, void **private_data);
+      int (*reloadModel) (const GstTensorFilterProperties * prop, void **private_data);
       /**< Optional. tensor_filter.c will call it when a model property is newly configured. Also, 'is-updatable' property of the framework should be TRUE. This function reloads a new model specified in the 'prop' argument. Note that it requires extra memory size enough to temporarily hold both old and new models during this function to hide the reload overhead.
        *
        * @param[in] prop read-only property values
        * @param[in/out] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer. Normally, close() frees private_data and set NULL.
        * @return 0 if ok. < 0 if error.
        */
 
-  int (*checkAvailability) (accl_hw hw);
+      int (*checkAvailability) (accl_hw hw);
       /**< Optional. Check if the provided hardware accelerator is supported. This check is static or dynamic based on framework support. Positive response of this check does not guarantee successful running of model with this accelerator. The static check can be performed without opening the framework.
        *
        * @param[in] hw backend accelerator hardware
        * @return 0 if supported. -errno if not supported.
        */
+    };
+
+    /**
+     * @brief Tensor_Filter Subplugin definition Version 2
+     */
+    struct /** _GstTensorFilterFramework_v2 */
+    {
+      int (*getFrameworkInfo) (void **private_data, GstTensorFilterFrameworkInfo *fw_info);
+      /**< Mandatory callback. Get the framworks statically determined info. Argument 'private_data' can be NULL. If provided 'private_data' is not NULL, then some info, such as 'allocate_in_invoke', can be updated based on the model being used (inferred from the 'private_data' provided). This updated info is useful for custom filter, as some custom filter's ability to support 'allocate_in_invoke' depends on the opened model.
+       *
+       * @param[in] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer. This parameter can be NULL.
+       * @param[out] fw_info struct to hold frameworks info. Must be allocated by the caller (return value).
+       * @return 0 if OK. non-zero if error.
+       */
+
+      int (*getModelInfo) (const GstTensorFilterProperties * prop,
+          void **private_data, model_info_ops ops, GstTensorsInfo *in_info,
+          GstTensorsInfo *out_info);
+      /**< Mandatory callback. Gets the model related tensor info.
+       * If ops == GET_INPUT_INFO, in_info would contain the input tensor info.
+       * If ops == GET_OUTPUT_INFO, out_info would contain the output tensor info.
+       * If ops == SET_INPUT_INFO, in_info would contain the provided input tensor info, and out_info would contain the updated output tensor info.
```

Comment:
It appears that you were right about this yesterday. I looks like that we are ok to merge GET-INPUT and GET-OUTPUT as a single op.

Reply:
Merged into a single OP and renamed them,